Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLReduction.py

```diff
@@ -491,6 +551,11 @@ def _apply_beam(self, ws, beam_ws):
             AddSampleLog(Workspace=ws, LogName='BeamCenterX', LogText=str(beam_x), LogType='Number')
             AddSampleLog(Workspace=ws, LogName='BeamCenterY', LogText=str(beam_y), LogType='Number')
             MoveInstrumentComponent(Workspace=ws, X=-beam_x, Y=-beam_y, ComponentName='detector')
+            try:
+                beam_width_x = beam_ws.getRun().getLogData('BeamWidthX').value
```

Comment:
Same here, try is excessive.

Reply:
Can you remove this try and check in dictionary as well?
Also here you are not setting the unit in the destination workspace.